File: Interpreter/ConvertFormatFun.java




class ConvertFormatFun (CalculateFunction):
	ConvertFormatFun(String name, ComplexFormatter formatter){
		funName = name
		complexFormatter = formatter
	}
	
	String getName(){
		return funName
	}
	
	boolean execute(self, paraList, context){
		if(len(paraList) != 1)
	    {
	        context.setErrorMessage(self.getName(), R_string.error_invalid_parameter_count)
	        return False
	    }
	    Complex para = paraList.getFirst()
	    context.setCurrentResult(para)
	    context.setFormatter(complexFormatter)
	    return self.checkResult(context)
	}
	
	private String funName
	private ComplexFormatter complexFormatter
}

File: Interpreter/ConvertFormatFun.py
from Interpreter.CalculateFunction import *


class ConvertFormatFun(CalculateFunction):
	def __init__(self, name, formatter):
		self.funName = name
		self.complexFormatter = formatter

	def get_name(self):
		return self.funName

	def execute(self, paraList):
		if len(paraList) != 1:
			context.setErrorMessage(self.get_name(), R_string.error_invalid_parameter_count)
			return False

		para = paraList[0]
		context.setCurrentResult(para)
		context.setFormatter(self.complexFormatter)
		return self.checkResult(context)
